controllers/forwarding.js

```diff
@@ -1,9 +1,23 @@
-const {asyncHandler} = require('./utils.js');
+const axios = require('axios');
 
-// takes a remote url for foreign service for deployment and a local port in development
+// takes a string for the env var name for deployment and a portNumber in development
 // returns endpoint that forwards request to either remote or local args
-exports.forwardingController = (remote = 'forward requres here', local ='fallback port if local') => (req, res, next) => {
-  //TODO handle forwarding, body, headers, routes
-  console.log('forwarding controller')
-  res.json(`${req.method}@${req.originalUrl} works. remote: ${ remote } local: ${ local }`); // test to make sure it works
-}
\ No newline at end of file
+exports.forwardingController = (envVarName = 'ENV var for service', portNumber ='port') => async (req, res, next) => {
+  // check environment, if prod use remote url env var, else use localhost port to make base of url
+  const serviceUrl = process.env.NODE_ENV === 'production' ? process.env[envVarName] : `http://localhost:${portNumber}`;
+
+  let { originalUrl, body, method, headers } = req;
+  method = method.toLowerCase();
+  // combine the serviceUrl with the matched path to get request url
+  const url = `${serviceUrl}${originalUrl}`;
+
+  // seperate get from other requests, make req forwarding body and headers and return response to client
+  let response;
+  if (method === 'get') {
+    response = await axios.get(url, {headers});
+  } else {
+    response = await axios[method](url, body, {headers});
+  }
+
+  res.send(response.data);
```

Comment:
One potential improvement here would be to destructure right when you're making your axios call:

**Meaning:**

_let response;
if (method === 'get') {
  response = await axios.get(url, {headers});
} else {
  response = await axios[method](url, body, {headers});
}_

res.send(response.data);

**Becomes:**

_if (method === 'get') {
  const { data } = await axios.get(url, {headers});
  res.send(data);
} else {
  const { data } = await axios[method](url, body, {headers});
  res.send(data);
}_

Really doesn't matter though and comes down to preference

Reply:
good call. i just added the .data in as a fix when i realized i was returning the axios metadata. will correct in the next pr